src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -692,7 +692,7 @@ class DetailsCard extends React.Component {
 
     // Data Center
     const dataCenterId = (clusters && clusters.getIn([clusterId, 'dataCenterId']))
-    const dataCenter = dataCenters && dataCenters.find(v => v.id === dataCenterId)
+    const dataCenter = dataCenters && dataCenters[dataCenterId]
```

Comment:
consider `const dataCenterName = dataCenters?.[dataCenterId]?.name ?? msg.notAvailable()`

Reply:
Since the render handles an `undefined` value for `dataCenter`, doing this would be all that is needed:
```suggestion
    const dataCenter = dataCenters?.[dataCenterId]
```

but that's really just a simple style change at that point.  IDK if it makes sense to convert to optional chaining in these cases for everythign we touch going forward or to just leave them in place.